panic when deploying R content without an existing renv.lock file

<!-- Please provide the following information to help us diagnose the issue, removing any sections that are not applicable. Thank you for your help! -->

## Problem statement

When I attempt to deploy an R shiny application without an existing renv.lock file, Publisher panics when trying to generate the lockfile.

## Steps to reproduce

Attempt to deploy this application:

```r
library("shiny")
library("bslib")
library("emoji")

ui <- page_fillable(
  layout_columns(
    card(
      card_header("Emoji selection"),
      selectizeInput(
        inputId = "name",
        label = "Emoji Name",
        choices = NULL
      ),
      textOutput(outputId = "emoji")
    )
  )
)

server <- function(input, output, session) {
  updateSelectizeInput(
    session,
    "name",
    choices = names(emoji_name),
    server = TRUE
  )
  output$emoji <- renderText({
    req(input$name)
    emoji_name[[input$name]]
  })
}

shinyApp(ui = ui, server = server)
```

**Expected behavior:** <!-- What should have happened? -->

Application deploys to Connect

**Actual behavior:** <!-- What actually happened? -->

Publisher appears to hang. Output logs show a panic.

https://github.com/user-attachments/assets/c6250fe0-7372-4af9-b2f1-b9158fc2210d

## System Details

## System Information

- **Posit Publisher Version**: 1.22.0
- **Platform**: darwin arm64
- **IDE**: Positron 1.103.0

## Publishing information

- **Content type:** R shiny app

<!--
If applicable: find this in Connect on System → Info tab → under System Info. See https://docs.posit.co/connect/admin/server-management/#determine-version-installed.

**Connect version:** e.g., 2025.09.0
-->

<!--
If applicable: find this in Connect Cloud at https://connect.posit.cloud/my-account.

**Connect Cloud account name:** e.g., my-account
-->

## Publisher output and logs

```
time=2025-10-23T12:43:51.263-04:00 level=INFO msg="Loading packages from renv.lock" localId=7cxo8qIqNVM2Atti event_op=publish/getRPackageDescriptions lockfile=/var/folders/kz/qtbxy1fx02n6vdc49y53lbg00000gn/T/publisher-renv-807939216/renv.lock
time=2025-10-23T12:43:51.263-04:00 level=DEBUG msg="Collecting manifest R packages" localId=7cxo8qIqNVM2Atti event_op=publish/getRPackageDescriptions lockfile=/var/folders/kz/qtbxy1fx02n6vdc49y53lbg00000gn/T/publisher-renv-807939216/renv.lock
time=2025-10-23T12:43:51.263-04:00 level=DEBUG msg="Reading lockfile for package manifest generation" localId=7cxo8qIqNVM2Atti lockfile=/var/folders/kz/qtbxy1fx02n6vdc49y53lbg00000gn/T/publisher-renv-807939216/renv.lock
time=2025-10-23T12:43:51.263-04:00 level=DEBUG msg="Processing packages from lockfile" localId=7cxo8qIqNVM2Atti package_count=54
panic: runtime error: slice bounds out of range [:4] with length 3

goroutine 856 [running]:
github.com/posit-dev/publisher/internal/inspect/dependencies/renv.isURL({0x14000a5ce43?, 0x1400097cf90?})
	/home/runner/work/publisher/publisher/internal/inspect/dependencies/renv/lockfile.go:104 +0xa4
github.com/posit-dev/publisher/internal/inspect/dependencies/renv.resolveRepoAndSource(0x1400097c810, {0x14000a5ce43, 0x3}, {0x14000a5cd16, 0xa})
	/home/runner/work/publisher/publisher/internal/inspect/dependencies/renv/manifest_packages_from_lockfile.go:253 +0x5c
github.com/posit-dev/publisher/internal/inspect/dependencies/renv.(*LockfilePackageMapper).GetManifestPackagesFromLockfile(0x140006a4c60, {{{0x140007221e0?, 0x20?}, {0x101150020?, 0x1014eb0a0?}}})
	/home/runner/work/publisher/publisher/internal/inspect/dependencies/renv/manifest_packages_from_lockfile.go:90 +0x420
github.com/posit-dev/publisher/internal/inspect/dependencies/renv.(*LockfilePackageMapper).GetManifestPackages(0x10113c600?, {{{0x100f349fa?, 0x3?}, {0x140003e0140?, 0x10114fb88?}}}, {{{0x140007221e0?, 0x1010b0680?}, {0x101150020?, 0x1014eb0a0?}}}, {0x0?, ...})
	/home/runner/work/publisher/publisher/internal/inspect/dependencies/renv/manifest_packages_from_lockfile.go:41 +0x38
github.com/posit-dev/publisher/internal/publish.(*defaultPublisher).getRPackagesWithPath(0x14000c325a0, 0x1)
	/home/runner/work/publisher/publisher/internal/publish/r_package_descriptions.go:87 +0x598
github.com/posit-dev/publisher/internal/publish.(*defaultPublisher).createManifest(0x14000c325a0)
	/home/runner/work/publisher/publisher/internal/publish/manifest.go:37 +0x338
github.com/posit-dev/publisher/internal/publish.(*defaultPublisher).doPublish(0x14000c325a0)
	/home/runner/work/publisher/publisher/internal/publish/publish.go:277 +0x298
github.com/posit-dev/publisher/internal/publish.(*defaultPublisher).PublishDirectory(0x14000c325a0)
	/home/runner/work/publisher/publisher/internal/publish/publish.go:239 +0x2bc
github.com/posit-dev/publisher/internal/services/api.RouterHandlerFunc.PostDeploymentHandlerFunc.func31.1()
	/home/runner/work/publisher/publisher/internal/services/api/post_deployment.go:142 +0x118
created by github.com/posit-dev/publisher/internal/services/api.RouterHandlerFunc.PostDeploymentHandlerFunc.func31 in goroutine 964
	/home/runner/work/publisher/publisher/internal/services/api/post_deployment.go:134 +0x120c
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -24,6 +24,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - When a package has a present `RemoteSubdir`, the field `GithubSubdir` is added too
   to avoid issues with previous versions of Packrat not considering `RemoteSubdir` for
   packages hosted in Github. (#3194)
+- Projects referencing packages with RemoteRepos by name and not URL do not hang when deploying. (#3209)
 
 # Changed
 

--- a/internal/inspect/dependencies/renv/lockfile.go
+++ b/internal/inspect/dependencies/renv/lockfile.go
@@ -107,7 +107,7 @@ func ValidateModernLockfile(lockfile *Lockfile) error {
 // and LockfilePackageMapper (lockfile-only approach) must produce identical output formats regardless
 // of whether packages reference repositories by name ("CRAN") or URL ("https://cloud.r-project.org").
 func isURL(s string) bool {
-	return len(s) > 0 && (s[0:4] == "http" || s[0:3] == "ftp") && strings.Contains(s, "://")
+	return len(s) > 0 && strings.Contains(s, "://") && (s[0:3] == "ftp" || s[0:4] == "http")
 }
 
 // Example package installed from CRAN

--- a/internal/inspect/dependencies/renv/manifest_packages_from_lockfile_test.go
+++ b/internal/inspect/dependencies/renv/manifest_packages_from_lockfile_test.go
@@ -427,8 +427,9 @@ func (s *LockfilePackageMapperSuite) TestBioconductor_LockfileCompatibility() {
 }
 
 func (s *LockfilePackageMapperSuite) TestRSPMRepositoryHandling() {
-	// Test that RSPM repository references are properly handled when not in repositories section
-	// but available through RemoteRepos fields
+	// Test repository references available through RemoteRepos fields:
+	// - RSPM properly handled when not in repositories section
+	// - Fictional turbopackages as RemoteRepos by name, URL infered from Repositories
 	lockfileContent := `{
 		"R": {
 			"Version": "4.3.3",
@@ -436,6 +437,10 @@ func (s *LockfilePackageMapperSuite) TestRSPMRepositoryHandling() {
 				{
 					"Name": "CRAN",
 					"URL": "https://cloud.r-project.org"
+				},
+				{
+					"Name": "turbopackages",
+					"URL": "https://turbopackages.org/latest"
 				}
 			]
 		},
@@ -456,6 +461,22 @@ func (s *LockfilePackageMapperSuite) TestRSPMRepositoryHandling() {
 					"R"
 				],
 				"Hash": "470851b6d5d0ac559e9d01bb352b4021"
+			},
+			"turbobear": {
+				"Package": "turbobear",
+				"Version": "99.99.99",
+				"Source": "Repository",
+				"RemoteType": "standard",
+				"RemotePkgRef": "turbobear",
+				"RemoteRef": "turbobear",
+				"RemoteRepos": "turbopackages",
+				"RemoteReposName": "turbopackages",
+				"RemotePkgPlatform": "x86_64-apple-darwin20",
+				"RemoteSha": "99.99.99",
+				"Requirements": [
+					"R"
+				],
+				"Hash": "990851b6d5d0ac559e9d01bb352b4021"
 			}
 		}
 	}`
@@ -480,6 +501,17 @@ func (s *LockfilePackageMapperSuite) TestRSPMRepositoryHandling() {
 	s.Equal("RSPM", r6Pkg.Source)
 	// Check version from description
 	s.Equal("2.5.1", r6Pkg.Description["Version"])
+
+	// Verify that the turbobear package was processed successfully
+	s.Contains(manifestPackages, "turbobear")
+	turbobearPkg := manifestPackages["turbobear"]
+
+	// turbopackages should be resolved through the RemoteRepos field
+	s.Equal("https://turbopackages.org/latest", turbobearPkg.Repository)
+	s.Equal("turbopackages", turbobearPkg.Source)
+
+	// Check version from description
+	s.Equal("99.99.99", turbobearPkg.Description["Version"])
 }
 
 func (s *LockfilePackageMapperSuite) TestRSPMRepositoryHandling_MissingRemoteRepos() {

